File: source/downloads/code/LeetCode/Python/Contains-Duplicate2.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        hash_map = dict()
        for i in range(len(nums)):
            idx = hash_map.get(nums[i])
            if idx >= 0 and k>= i-idx:
                    return True
            hash_map[nums[i]] = i
        return False

File: downloads/code/LeetCode/Java/Contains-Duplicate2.java
public class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> hash_map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++){
            if (hash_map.containsKey(nums[i])){
                if (i-hash_map.get(nums[i]) <= k) return true;
            }
            // 若nums[i]存在,但是i-hash_map.get(nums[i])>k, 也要更新hash_map            
            hash_map.put(nums[i],i);
        }
        return false;     
    }
}
